homeassistant/components/yale_smart_alarm/binary_sensor.py

```diff
@@ -24,16 +56,58 @@ async def async_setup_entry(
     ]
 
     async_add_entities(
-        YaleBinarySensor(coordinator, data) for data in coordinator.data["door_windows"]
+        YaleDoorSensor(coordinator, data) for data in coordinator.data["door_windows"]
     )
 
+    async_add_entities(
+        YaleProblemSensor(coordinator, description) for description in SENSOR_TYPES
+    )
 
-class YaleBinarySensor(YaleEntity, BinarySensorEntity):
-    """Representation of a Yale binary sensor."""
+
+class YaleDoorSensor(YaleEntity, BinarySensorEntity):
+    """Representation of a Yale door sensor."""
 
     _attr_device_class = BinarySensorDeviceClass.DOOR
 
     @property
     def is_on(self) -> bool:
         """Return true if the binary sensor is on."""
         return self.coordinator.data["sensor_map"][self._attr_unique_id] == "open"
+
+
+class YaleProblemSensor(CoordinatorEntity, BinarySensorEntity):
+    """Representation of a Yale problem sensor."""
+
+    entity_description: BinarySensorEntityDescription
+    coordinator: YaleDataUpdateCoordinator
+
+    def __init__(
+        self,
+        coordinator: YaleDataUpdateCoordinator,
+        entity_description: BinarySensorEntityDescription,
+    ) -> None:
+        """Initiate Yale Problem Sensor."""
+        super().__init__(coordinator)
+        self.entity_description = entity_description
+        self._attr_name = (
+            f"{coordinator.entry.data[CONF_NAME]} {entity_description.name}"
+        )
+        self._attr_unique_id = f"{coordinator.entry.entry_id}-{entity_description.key}"
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, coordinator.entry.data[CONF_USERNAME])},
+            manufacturer=MANUFACTURER,
+            model=MODEL,
+            name=coordinator.entry.data[CONF_NAME],
+            connections={
+                (CONNECTION_NETWORK_MAC, coordinator.data["panel_info"]["mac"])
+            },
+            sw_version=coordinator.data["panel_info"]["version"],
+        )
```

Comment:
Is there an opportunity to make this a bit more DRY?

Reply:
I made a base entity for alarm too so saving a few lines